components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx

```diff
@@ -160,25 +208,35 @@ export default class PostAttachmentOpenGraph extends React.PureComponent {
             imageUrl && renderingForLargeImage === this.state.hasLargeImage &&
             (!renderingForLargeImage || (renderingForLargeImage && this.props.isEmbedVisible))
         ) {
-            if (renderingForLargeImage) {
-                const imageDimensions = getFileDimensionsForDisplay(metadata.images && metadata.images[imageUrl], MAX_DIMENSIONS_LARGE_IMAGE);
-
-                element = (
-                    <img
-                        className={'attachment__image attachment__image--opengraph large_image'}
-                        src={imageUrl}
-                        {...imageDimensions}
-                    />
-                );
-            } else {
-                const imageDimensions = getFileDimensionsForDisplay(metadata.images && metadata.images[imageUrl], MAX_DIMENSIONS_SMALL_IMAGE);
-                element = this.wrapInSmallImageContainer(
-                    <img
-                        className={'attachment__image attachment__image--opengraph'}
-                        src={imageUrl}
-                        {...imageDimensions}
-                    />
-                );
+            if (this.props.post.metadata || (!this.props.post.metadata && this.state.imageLoaded)) {
+                if (renderingForLargeImage) {
+                    const imageDimensions = getFileDimensionsForDisplay(metadata.images && metadata.images[imageUrl], MAX_DIMENSIONS_LARGE_IMAGE);
+
+                    element = (
+                        <img
+                            className={'attachment__image attachment__image--opengraph large_image'}
+                            src={imageUrl}
+                            {...imageDimensions}
+                        />
+                    );
+                } else {
+                    const imageDimensions = getFileDimensionsForDisplay(metadata.images && metadata.images[imageUrl], MAX_DIMENSIONS_SMALL_IMAGE);
+                    element = this.wrapInSmallImageContainer(
+                        <img
+                            className={'attachment__image attachment__image--opengraph'}
+                            src={imageUrl}
+                            {...imageDimensions}
+                        />
+                    );
+                }
+            } else if (!this.state.imageLoaded) {
```

Comment:
Similarly, this doesn't need the if part since we would've gone into the above branch if `imageLoaded` was true
```suggestion
            } else {
```

Reply:
This was a half baked cleanup job from really old code. Made changes. Should be better now